### PR TITLE
Change all clarifications in thread

### PR DESCRIPTION
**Pull request by vmcj**
By always finding the first in thread and changing the replies afterwards. Verified that this also works via the overview clarification page.

Partial fix for: https://github.com/DOMjudge/domjudge/issues/1785

### CODE REVIEW

File: webapp/src/Controller/Jury/ClarificationController.php [hunk 1]
```diff
@@ -352,7 +352,17 @@ public function changeQueueAction(Request $request, int $clarId): Response
...
             $queue = null;
         }
-        $clarification->setQueue($queue);
+
+        // Either this is the first in a thread (of 1 or more) or this 2nd or more.
+        $firstClarification = $clarification->getInReplyTo();
+        $curClarification = $clarification;
+        if ($firstClarification) {
+            $curClarification = $firstClarification;
+        }
```
Comment (nickygerritsen) [inline webapp/src/Controller/Jury/ClarificationController.php:361]:
```suggestion
        $curClarification = $clarification->getInReplyTo() ?? $clarification;
```

Or maybe `?:` if it doesn't return `null` but `false`

Comment (vmcj) [reply webapp/src/Controller/Jury/ClarificationController.php:361]: `public function getInReplyTo(): ?Clarification`

So `??` should be good?

**Review by thijskh — APPROVED** [review]

File: webapp/src/Controller/Jury/ClarificationController.php [hunk 2]
```diff
@@ -352,7 +352,13 @@ public function changeQueueAction(Request $request, int $clarId): Response
         if ($queue === "") {
             $queue = null;
         }
-        $clarification->setQueue($queue);
+
+        // Either this is the first in a thread (of 1 or more) or this 2nd or more.
```
Comment (eldering) [inline webapp/src/Controller/Jury/ClarificationController.php:356]: This comment is not really clear to me: `or this 2nd or more.`

Comment (vmcj) [reply webapp/src/Controller/Jury/ClarificationController.php:356]: Removed the comment as Nicky his fix made this easy enough to read in code.

**Pull request merged**